test/framework/src/main/java/org/elasticsearch/cluster/routing/TestShardRouting.java

```diff
@@ -307,16 +310,20 @@ public static UnassignedInfo randomUnassignedInfo(String message) {
             if (randomBoolean()) {
                 delayed = true;
             }
-            lastAllocatedNodeId = randomAlphaOfLength(10);
+            lastAllocatedNodeId = randomIdentifier();
         }
         int failedAllocations = reason == UnassignedInfo.Reason.ALLOCATION_FAILED ? 1 : 0;
+        long unassignedTimeNanos = randomLongBetween(
+            1640991600000L, // 01-01-2022
+            1672527600000L // 01-01-2023
```

Comment:
Hmm these are in milliseconds not nanoseconds (and they're midnight CET I think, not UTC). Do we need to specify the range like this or can we just say `randomNonNegativeLong()`?

Reply:
I originally tried `randomNonNegativeLong()` however number of tests failed as unassigned time appeared in the future.
I think we need the range, however it might not be static, possibly last month or year within given timezone or UTC

Reply:
Ok, we can deal with that I think, but the code as written here yields very unrealistic timestamps in practice. The dates are not in 2022 as the comments suggest, they're some time in early 1970.

I think we should detach the (node-relative) nanoseconds timestamp from the (absolute) milliseconds timestamp. The node-relative one shouldn't matter much, although if we need it to be in the past then we can wait for `System::nanotime` to be at least 1e9 (at most 1s) and then pick a timestamp in `[0, 1e9]`. The millis one can be based on a fixed absolute date, but shouldn't need a tighter lower bound than 0 IMO.

Reply:
I was thinking about the generator behavior. We could make the test to generate offset to the current time. This way the generated data is not going to be the same, however it will not become a years in the past over the time, making it easier to test small and long time intervals if needed.

Will also update the nano time.